api/types/generatorargs.go

```diff
@@ -21,4 +21,7 @@ type GeneratorArgs struct {
 
 	// KvPairSources for the generator.
 	KvPairSources `json:",inline,omitempty" yaml:",inline,omitempty"`
+
```

Comment:
We should be able to control the behavior w.r.t. the global -- e.g. if there is per-resource options AND global options in the same kustomization.yaml, should they be unioned / merged, or only take the local.

Would suggest merging with global by default with the option to not use the global.

Reply:
Definitely, currently it will merge them.